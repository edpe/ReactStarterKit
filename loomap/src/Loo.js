import React from 'react';

import LooMap from './LooMap';

const Loo = (props) => {
  let coords = props.loo.geometry.coordinates.slice(0).reverse();
  return (
    <div>
      <h3>{props.loo.properties.name}</h3>
      <LooMap
        center={coords}
        zoom={15}
        height="200px"
        geojson={props.loo}
        fixed={true}
        />
    </div>
  );
}

export default Loo;
