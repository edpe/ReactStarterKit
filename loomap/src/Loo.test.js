import React from 'react';
import Loo from './Loo';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';

const data = {
	"_id": "5a031106a762310011e94c4f",
	"properties": {
		"name": "Piata Unitii"
	},
	"geometry": {
		"type": "Point",
		"coordinates": [23.589, 46.770]
	},
	"type": "Feature",
	"distance": 245
};

describe('Loo', () => {
  it('Renders a heading and a Map', () => {
    const div = global.document.createElement('div');
    global.document.body.appendChild(div);
    const component = mount(
      <Loo loo={data}/>,
      {
        attachTo: div
      }
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
