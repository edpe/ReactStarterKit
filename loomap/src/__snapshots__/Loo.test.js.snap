// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loo Renders a heading and a Map 1`] = `
<Loo
  loo={
    Object {
      "_id": "5a031106a762310011e94c4f",
      "distance": 245,
      "geometry": Object {
        "coordinates": Array [
          23.589,
          46.77,
        ],
        "type": "Point",
      },
      "properties": Object {
        "name": "Piata Unitii",
      },
      "type": "Feature",
    }
  }
>
  <div>
    <h3>
      Piata Unitii
    </h3>
    <LooMap
      center={
        Array [
          46.77,
          23.589,
        ]
      }
      fixed={true}
      geojson={
        Object {
          "_id": "5a031106a762310011e94c4f",
          "distance": 245,
          "geometry": Object {
            "coordinates": Array [
              23.589,
              46.77,
            ],
            "type": "Point",
          },
          "properties": Object {
            "name": "Piata Unitii",
          },
          "type": "Feature",
        }
      }
      height="200px"
      zoom={15}
    >
      <Map
        center={
          Array [
            46.77,
            23.589,
          ]
        }
        dragging={false}
        keyboard={false}
        scrollWheelZoom={false}
        style={
          Object {
            "height": "200px",
          }
        }
        touchZoom={false}
        zoom={15}
        zoomControl={false}
      >
        <div
          style={
            Object {
              "height": "200px",
            }
          }
        >
          <TileLayer
            attribution="Â© OpenStreetMap"
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          />
          <GeoJSON
            data={
              Object {
                "_id": "5a031106a762310011e94c4f",
                "distance": 245,
                "geometry": Object {
                  "coordinates": Array [
                    23.589,
                    46.77,
                  ],
                  "type": "Point",
                },
                "properties": Object {
                  "name": "Piata Unitii",
                },
                "type": "Feature",
              }
            }
          />
        </div>
      </Map>
    </LooMap>
  </div>
</Loo>
`;
